### ISSUE TITLE
Can not find sst file when executing query

### ISSUE THREAD

**Issue by MachaelLee** [CeresDB/ceresdb#534]
### Describe this problem

Error message is:
```
2023-01-03 20:23:31.891 ERRO [server/src/grpc/storage_service/mod.rs:258] Failed to handle request, mod:query, handler:handle_query, err:Rpc error, code:500, message:Failed to execute interpreter, query:SELECT data_source,day_of_growth,gmt_create,gmt_modified,id,net_value,net_value_date,profit_seven_days,profit_ten_thousand,restored_net_value,symbol,total_net_value FROM `quotresearch_fund_quotation_v1` WHERE (is_dfd_del=0 OR is_dfd_del IS NULL) AND net_value_date>=1664972608726 AND net_value_date<=1672748608726, cause:Failed to execute select, err:Failed to execute logical plan, err:Failed to collect record batch stream, err:Stream error, msg:convert from arrow record batch, err:External error: Stream error, msg:read record batch, err:Failed to read data from the sub iterator, err:PullRecordBatch { source: DecodeRecordBatch { source: ParquetError { source: General("Failed to fetch ranges from object store, err:Generic Aliyun error: Failed to get range of object at path:ceresdbfin/0/172/1969.sst, err:error sending request for url (http://antsys-ceresdbfin.cn-shanghai-alipay-internal.oss-alipay.aliyuncs.com/ceresdbfin/0/172/1969.sst?): error trying to connect: dns error: failed to lookup address information: Name or service not known"), backtrace: Backtrace(   0: <snafu::backtrace_shim::Backtrace as snafu::GenerateBacktrace>::generate
```

And there is no file 'ceresdbfin/0/172/1969.sst' in oss:
![Uploading image.png…]()

### Steps to reproduce

None

### Expected behavior

Query result

### Additional Information

Maybe it is related to file merging：
```
2023-01-03 20:23:31.992 INFO [analytic_engine/src/row_iter/merge.rs:850] Merge iterator dropped, table_id:TableId(172), request_id:598719, metrics:Metrics { num_memtables: 1, num_ssts: 9, sst_ids: [820, 1969, 2002, 2034, 2098, 2179, 2184, 2191, 2199], times_fetch_rows_from_one: 0, total_rows_fetch_from_one: 0, times_fetch_row_from_multiple: 368838, duration_since_create: 1.022840061s, init_duration: 49.685179ms, scan_duration: 167.588723ms, scan_count: 422 }, iter_options:IterOptions { batch_size: 500, sst_background_read_parallelism: 8 },
```

**Comment by jiacai2050** [comment]
I think this issue is similar with https://github.com/CeresDB/ceresdb/issues/427.

The problem is that we have no snapshot guarantee for a query. 

When a query begin, we should make a snapshot of files it required, and KEEP them duration query execution, only after query finish, we can delete files used in this query.

**Comment by jiacai2050** [comment]
Closed in #699
